### DISCUSSION TITLE
Escaping characters: msg("& string"... -> <source>&amp; string</source> -> translation prints as &amp;

### DISCUSSION THREAD

**Discussion by pika11** [lit/lit#2715] [Q&A]
When `lit-localize` extracts to .xlf it scapes some characters from source. Then when translated it gets printed as "&amp; translated string"

Is there an option to turn off `escapeHtml` when getting the source, or alternative to unescape when printing?

**Comment by aomarks** [comment]
Could you give an example? And when you say "translated", do you mean in output generated by Lit Localize (runtime or transform mode?) -- or somewhere else, like in the UI for the translation tool/service you are using?

**Reply by pika11** [reply]
Sure, and thank you for your quick reply. This is in transform mode:

- ts file:
`msg("Terms & Conditions")`

- output in locale xlf file:
```
<trans-unit id="">
  <source>Terms &amp; Conditions</source>
  <target state="final">Translated:Terms &amp; Translated:Conditions</target>
</trans-unit>
```
- When browsing the site on that locale it'll print as `Translated:Terms &amp; Translated:Conditions`


Question here would be if when rendering this, 
1. Could we change &amp; to the original `&`. 
2. Or alternatively if the <source></source> in .xlf files could be preserved as `&`

I think both options could solve our issue.

**Reply by pika11** [reply]
markdown formatting issue one second - updated

**Reply by pika11** [reply]
@aomarks does the example above makes sense? or let me know if i should explain a bit better. :thankyou:

**Reply by pika11** [reply]
Hey @aomarks sorry for the ping, want to resurface this

**Reply by aomarks** [reply]
Sorry, haven't had a chance to investigate yet. It could potentially be a bug in our XML serialization logic.

I see that you have a `state="final"` attribute in your `<target>` node. That indicates that this XML file was generated by another tool, since `lit-localize extract` doesn't generate the `state` attribute. Can you confirm that the XML looks the same when you run `lit-localize extract`?

**Reply by pika11** [reply]
No worries just want to check if is something on our side or in lit's.

Yeah indeed `lit-localize extract` doesn't generate `state="final"` but other than that everything remains as the example above.

**Comment by pika11** [comment]
@aomarks okay This is my understanding:
- https://github.com/lit/lit/blob/9fc5a039dc2b701ac9dbaaea278668172915c80b/packages/localize-tools/src/typescript.ts#L89 

This `ts` helper scapes the string when getting it from the source code but it's never changed back when reading from the built locale files then it just renders with the scape format. 

source -> extract -> `scape html` -> generate xml -> translate xml -> build locale files -> render 

IMO there should be an opposite step to the scape html function before the build (for the build files to be the same as the source only translated) or just before render. 

Let me know if there's something I might be missing here. Also let me know if you need help with this or the other reported issue https://github.com/lit/lit/issues/2713 . Glad to help.
